installer/scripts/service_control

```diff
@@ -302,167 +518,158 @@ disable_omsagent_service()
     local omsagent_service=${systemd_dir}/$OMSAGENT_WS.service
     local omsagent_initd=/etc/init.d/$OMSAGENT_WS
     if [ -f $omsagent_service ]; then
-        echo "Unconfiguring OMS agent (systemd) service ..."
+        echo "INFO:  Unconfiguring OMS agent (systemd) service $WORKSPACE_ID ..."
         /bin/systemctl -q disable $OMSAGENT_WS
         rm -f $omsagent_service
         /bin/systemctl daemon-reload
+        WS_STATUS=$?
     elif [ -f $omsagent_initd ]; then
-        echo "Unconfiguring OMS agent service ..."
+        echo "INFO:  Unconfiguring OMS agent service $WORKSPACE_ID ..."
         if [ -f /usr/sbin/update-rc.d ]; then
             /usr/sbin/update-rc.d -f $OMSAGENT_WS remove
         elif [ -x /usr/lib/lsb/remove_initd ]; then
             /usr/lib/lsb/remove_initd $omsagent_initd
         elif [ -x /sbin/chkconfig ]; then
             /sbin/chkconfig --del $OMSAGENT_WS > /dev/null
         else
-            echo "Unrecognized Service Controller to unregister OMS Agent Service."
+            echo "FATAL:  Unrecognized Service Controller to unregister OMS Agent Service."
             exit 1
         fi
+        WS_STATUS=$?
 
         rm $omsagent_initd
     fi
     rm -f $CONF_DIR/.service_registered
+    return $WS_STATUS
 }
 
-check_ws_and_invoke()
-{
-    for ws_id in $NON_OMS_WS
-    do
-        ls -1 $ETC_DIR | grep -w ${ws_id} > /dev/null 2>&1
-        if [ $? -eq 0 ]; then
-            setup_variables ${ws_id}
-            $1
-        fi
-    done
-}
+# ## Interface Sub-Submethods
 
-start_all_omsagent()
+generateProcedureName()
```

Comment:
nit: name this using the same style as the rest of the method names in this file: `generate_procedure_name`